Model admin for scan results

chigby opened the pull request:
This pull request replaces the inline view of scan results that appears on the directory entry admin page with a separate Model Admin area. The reason for this is the inline scan result list becomes slow if there are a large number of results, and also can potentially break the edit functionality.

Instead of that, I've added a drop down button on the page listing for directory entries:

![image](https://user-images.githubusercontent.com/561931/101408025-c0c9a100-38a9-11eb-90f9-5e3515aaa3ee.png)

Fixes #777

chigby committed:
Add ModelAdmin for scan results

Features

1. Read only
2. Can inspect an individual result
3. Basic filters/searching only (we can add more if needed)

chigby committed:
Remove inline scan results from directory entry admin

If an entry has many results, this functionality is slow or breaks the
page completely.

SaptakS commented on directory/wagtail_hooks.py:
```diff
@@ -0,0 +1,87 @@
+@hooks.register('add_scan_results_button')
+def add_bundle_stats_button(page, page_perms, is_parent=False):
+    index_url = scanresult_modeladmin.url_helper.get_action_url('index')
+    results_url = index_url + '?securedrop__page_ptr_id__exact=' + str(page.pk)
```
I prefer `.format()`. In fact since we have updated to python3.7, we should even be able to use `f'{index_url}?securedrop__page_ptr_id__exact={page.pk}'`. Just makes better readability according to me.

Also, maybe we can add docstring to this hook and the ScanResultAdmin class as well?

SaptakS commented on directory/wagtail_hooks.py:
```diff
@@ -0,0 +1,87 @@
+        'grade',
+        'live',
+    )
+    search_fields = ('landing_page_url',)
```
Curious if we should allow also the `securedrop` field for searching?

SaptakS reviewed (commented):
Looks good to me. Left some comment on code basis. Can be merged after that.

Just to note here, the scan results are no longer visible in the edit page. One will have to click the scan result dropdown button from the listing page. Also, they can directly go to the Scan Results and search by entering the landing page url. I think this works and is a substantial improvement but might be something to note and remember. cc @eloquence

chigby committed:
Allow searching ScanResults by title of directory entry

chigby committed:
Add links to scan results in directory page editor

chigby commented:
Thanks for the feedback here.

1. Definitely prefer the `f''` string approach here. Need to get into the habit of using that!
2. Added some additional docstrings.
3. Allow searching by directory entry title
4. Added a helpful link section at the bottom of the DirectoryEntry editor (where the big list of all the results was previously), so now you can preview and quickly jump to the results no matter where you are in the admin:

![image](https://user-images.githubusercontent.com/561931/101499759-a213ea80-393b-11eb-8d2c-71c3aa0a7aec.png)

SaptakS reviewed (approved):
looks good to me.
